### PR TITLE
Optimizes Replace Members with IDs function

### PR DESCRIPTION
**Pull request by purujitgoyal**

### CODE REVIEW

File: openreview/tools.py [hunk 1]
```diff
@@ -796,38 +797,23 @@ def replace_members_with_ids(client, group):
...
 
-    results = concurrent_requests(classify_members, group.members)
+    member_profiles = get_profiles(client, group.members, as_dict=True)
 
-    for key, member in results:
-        if key == 'ids':
-            ids.append(member)
-        elif key == 'emails':
-            emails.append(member)
+    for member, profile in member_profiles.items():
```
Comment (melisabok) [inline openreview/tools.py:804]: could you start the iteration with `group.members` to make sure each member is replaced or not found error. 

```
for member in groups.members:
   profile = member_profiles.get(member)
   if profile:
          update_members.append(profile.id)
   else:
          if '@' in member:
                update_members.append(profile.id)
         else:
               raise openreview.OpenReviewException(f"Profile Not Found for {member}")
```

We also make sure they are replaced in the same order.

Comment (carlosmondra) [reply openreview/tools.py:804]: Why do they need to be replaced in the same order?

Comment (melisabok) [reply openreview/tools.py]: I think maintaining the order is better but I may be wrong.

**Review by carlosmondra — APPROVED** [review]

File: openreview/tools.py (hunk 1 above)
Comment (carlosmondra) [reply openreview/tools.py]: Well, they are now being replaced in order, so I think it's fine.

**Review by melisabok — APPROVED** [review]

**Pull request merged**